### PR TITLE
Ensure MongoSpark.toDF() uses the set connector

### PR DESCRIPTION
**Pull request by rozza**
SPARK-220

### CODE REVIEW

File: src/main/scala/com/mongodb/spark/MongoSpark.scala [hunk 1]
```diff
@@ -604,10 +607,7 @@ case class MongoSpark(sparkSession: SparkSession, connector: MongoConnector, rea
    * @return a DataFrame.
    */
   def toDF(schema: StructType): DataFrame = {
-    sparkSession.read.format("com.mongodb.spark.sql")
-      .schema(schema)
-      .options(readConfig.asOptions)
-      .load()
+    sparkSession.baseRelationToDataFrame(MongoRelation(toBsonDocumentRDD, Option(schema))(sparkSession.sqlContext).asInstanceOf[BaseRelation])
```
Comment (rozza) [inline src/main/scala/com/mongodb/spark/MongoSpark.scala]: This essentially shortcuts the Spark logic for getting the dataframe from a relation.

File: src/test/scala/com/mongodb/spark/NoSparkConfSpec.scala [hunk 2]
```diff
@@ -59,9 +61,28 @@ class NoSparkConfSpec extends RequiresMongoDB {
     ds.collect().toList should equal(characters)
   }
 
+  it should "use the passed in connector" in {
+    val writeConfig = WriteConfig(Map("uri" -> mongoClientURI, "database" -> databaseName, "collection" -> collectionName))
+    val documents = sc.parallelize((1 to 10).map(i => Document.parse(s"{test: $i}")))
+    documents.saveToMongoDB(writeConfig = writeConfig)
+
+    val connector = MongoConnector(CustomMongoClientFactory(mongoClientURI))
+    val readConfig = ReadConfig(Map("uri" -> "mongodb://example.com/", "database" -> databaseName, "collection" -> collectionName,
```
Comment (rozza) [inline src/test/scala/com/mongodb/spark/NoSparkConfSpec.scala]: By using the bad uri - I ensure that the passed in connector with its hardcoded uri is used.

File: src/test/scala/com/mongodb/spark/NoSparkConfSpec.scala [hunk 3]
```diff
@@ -59,9 +61,28 @@ class NoSparkConfSpec extends RequiresMongoDB {
...
+      "partitioner" -> "TestPartitioner$"))
+    val mongoSpark = MongoSpark.builder().sparkContext(sc).connector(connector).readConfig(readConfig).build()
+
+    mongoSpark.toRDD().collect().size should equal(10)
+    mongoSpark.toDF().count() should equal(10)
```
Comment (rozza) [inline src/test/scala/com/mongodb/spark/NoSparkConfSpec.scala]: Without the change this fails.

File: src/main/scala/com/mongodb/spark/MongoSpark.scala (hunk 1 above)
Comment (jyemin) [reply src/main/scala/com/mongodb/spark/MongoSpark.scala]: Ack

File: src/test/scala/com/mongodb/spark/NoSparkConfSpec.scala (hunk 3 above)
Comment (jyemin) [reply src/test/scala/com/mongodb/spark/NoSparkConfSpec.scala]: Because it uses the ReadConfig with "mongodb://example.com" URI?  If so... ack!

**Pull request merged**